Description: Fix FTBFS with glibc 2.26
Bug-Debian: https://bugs.debian.org/887633
Author: Axel Beckert <abe@debian.org>
Forwarded: http://lists.xymon.com/archive/2017-February/044370.html

--- a/client/logfetch.c
+++ b/client/logfetch.c
@@ -29,6 +29,7 @@
 #include <glob.h>
 #include <pwd.h>
 #include <grp.h>
+#include <stdint.h>
 
 /* Some systems do not have the S_ISSOCK macro for stat() */
 #ifdef SCO_SV
